buildSrc/src/main/groovy/Classpaths.groovy

```diff
@@ -220,4 +228,14 @@ class Classpaths {
         Configuration config = p.configurations.getByName(configName)
         addDependency(config, p.getDependencies().platform(GRPC_GROUP + ":grpc-bom:" + GRPC_VERSION))
     }
+
+    static void inheritProtobufPlatform(Project p, String configName = JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME) {
+        Configuration config = p.configurations.getByName(configName)
+        addDependency(config, p.getDependencies().platform(PROTOBUF_GROUP + ":protobuf-java:" + PROTOBUF_VERSION))
+    }
+
+    static void inheritBoringSslPlatform(Project p, String configName = JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME) {
```

Comment:
We should not have "platform" here or as part of dependencies. Our default should likely be runtimeOnly instead of implementation to match what our caller is currently doing.

Reply:
Discussed with you on slack, fixed/updated, PTAL.

Reply:
Without using platform, we get a compilation error:

```
> Task :server:compileJava FAILED
/a0/h/cfs/dh/oss3/deephaven-core/server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java:8: error: cannot find symbol
import com.google.common.collect.Streams;
                                ^
  symbol:   class Streams
  location: package com.google.common.collect
1 error

> Task :web-client-ui:uiRemoveContainer
Removing container with ID 'ui-container-41c6e9de-6c1e-42f1-ba1d-75265a18975a'.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':server:compileJava'.
> Compilation failed; see the compiler error output for details.
```

Reply:
I recovered "Platform" for protobuf; otherwise it doesn't compile.

PTAL, I am happy to do this differently but will need more specific guidance as for what to change to what.

Reply:
something else is wrong then, i'll take a look.

Reply:
It looks like we were "lucky" it was working with platform IMO, b/c that's *not* how it was setup before. It looks like ultimately there is some difference in the resolution of the transitive dependencies that is changing the version of guava that is available at compile time.

You can get a full list of the compile classpath for the server module using:

```
./gradlew server:dependencies --configuration compileClasspath
```

I compared the difference on `main` vs your branch with my suggested fixes to Classpaths, and looked at the diff wrt guava:

```
$ grep guava /tmp/main.txt 
|    |    |    |              \--- com.google.guava:guava:30.1-android -> 30.1.1-jre
|    |    |    |                   +--- com.google.guava:failureaccess:1.0.1
|    |    |    |                   +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
|    |    |    |    +--- com.google.guava:guava:30.1.1-jre (*)
$ grep guava /tmp/branch.txt                                                      
|    |    |    |              \--- com.google.guava:guava:31.0.1-android
|    |    |    |                   +--- com.google.guava:failureaccess:1.0.1
|    |    |    |                   +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
|    |    |    |    +--- com.google.guava:guava:30.1.1-jre -> 31.0.1-android (*)
```

The important thing being that main is resolving `guava:30.1.1-jre`, but your branch is resolving `31.0.1-android`. It likely has to do with how the arrow project or some other updated dependency has changed the structure of its transitive dependencies.

The good news is that the usage of guava/Streams in BarrageMessageProducer was likely an accidental import by @nbauernfeind - we can use the built-in Stream instead of guava Streams.

I've pushed the changes to my fork.

Reply:
https://github.com/jcferretti/deephaven-core/pull/3

Reply:
Pulled/merged.